pkg/rngd/cmd/rngd/main.go

```diff
@@ -14,7 +15,76 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+type rng struct {
+	rngName  string
+	rngDev   string
+	disabled bool
+	rngInit  xinit
+	rngRead  xread
+}
+
+type xinit func(ctx *rng) bool
+type xread func(ctx *rng) (uint64, error)
+
+// Check if hardware RNG device is valid
+func initEntropySource(ctx *rng) bool {
+	f, err := os.Open(ctx.rngDev)
+	defer f.Close()
+	if err != nil {
+		ctx.disabled = true
+		return false
+	}
+
+	// Try to read some data from the entropy source. If it doesn't
+	// return an error, assume it's ok to use
+	buf := make([]byte, 16)
+	n, err := f.Read(buf)
+	if err != nil || n != 16 {
+		ctx.disabled = true
+		return false
+	}
+
+	return true
+}
+
+// Read entropy from the hardware RNG device
+func readHwRNG(ctx *rng) (uint64, error) {
+	f, err := os.Open(ctx.rngDev)
+	defer f.Close()
+	if err != nil {
+		return 0, err
+	}
+	buf := make([]byte, 8)
+	_, err = f.Read(buf)
+	if err != nil {
+		return 0, err
+	}
+
+	r := binary.LittleEndian.Uint64(buf)
+
+	return r, nil
+}
+
+var entropySources = []rng{
+	// Hardware RNG device
+	{
+		rngName: "Hardware RNG Device",
+		rngDev:  "/dev/hwrng",
+		rngInit: initEntropySource,
+		rngRead: readHwRNG,
+	},
+	// TPM support will be here
```

Comment:
I believe tmp uses `/dev/hwrng` too, so this comment is not needed

Reply:
I believe you are right.

Reply:
OK, I will remote that.